cmd/gardener-apiserver/app/gardener_apiserver.go

```diff
@@ -132,9 +133,9 @@ func (o *Options) complete() error {
 	recommendedPluginOrder.Insert(allOrderedPlugins...)
 	o.Recommended.Admission.RecommendedPluginOrder = recommendedPluginOrder.List()
 
-	// TODO: Temporary deletionconfirmation it as disabled by default.
-	// Enable it once the old shoot deletion mechanics is removed.
-	o.Recommended.Admission.DefaultOffPlugins.Insert(deletionconfirmation.PluginName)
+	if !gardenfeatures.APIServerFeatureGate.Enabled(gardenfeatures.DeletionConfirmation) {
+		o.Recommended.Admission.DefaultOffPlugins.Insert(deletionconfirmation.PluginName)
```

Comment:
To clarify: Due to the fact that the `DeletionConfirmation` admission controller is not part of `o.Recommended.Admission.RecommendedPluginOrder` list you can only enable it if the feature gate is enabled **and** you put its name in the `--enable-admission-plugins` flag, is it?

Reply:
Everything that is in `o.Recommended.Admission.RecommendedPluginOrder` is **enabled** by default.
When this feature flag is **not enabled** then `DeletionConfirmation` plugin is put in the `DefaultOffPlugins` which means that it is **disabled**. 

You can **at any time** enable it explicitly with `--enable-admission-plugins`

Reply:
OK, once again:
I can enable it anytime (independent of whether the feature flag is set or not) with `--enable-admission-plugins`.
If I don't use `--enable-admission-plugins` but only set the feature flag, what happens then? Nothing?

Reply:
OK, understood, thank you.